### PR TITLE
Adding mini-group types

### PR DESCRIPTION
**Pull request by mhassans**
minigroup types added. One way of defining a minigroup is the usual minigroups that Samuel implemented which is the smallest possible minigroup we can have (hence called 'minimal' in the config). I added another minigroup type, 'bylayer', in which each layer is a minigroup. I also added 'bylayer_silicon_separated' in which silicon layers and scintillator layers are in different minigroups (in addition to be separated by layer)

### CODE REVIEW

File: config/default.yaml [hunk 1]
```diff
@@ -1,24 +1,27 @@
...
   # algorithm: genetic_alg      
   #initial_state: bestsofar
   initial_state: random
-  #initial_state: bestsofar_38.npy       
+  #initial_state: bundles_job_default.npy   
   random_seed: ~
   max_iterations: 100000 
+  #minigroup_type: 'bylayer_silicon_seprated'
+  #minigroup_type: 'bylayer'
+  minigroup_type: 'minimal'
```
Comment (snwebb) [inline config/default.yaml:24]: I would add these lines (minigroup_type), but remove the other changes to this file.

File: process.py [hunk 2]
```diff
@@ -302,42 +302,71 @@ def getlpGBTHists(data, module_hists):
...
-                minigroups[row['TPGId2']] = minigroups[row['TPGId1']]
-            else:
-                minigroups[row['TPGId1']] = counter
-                minigroups[row['TPGId2']] = counter
-
+            if (layer !=layers_connected_to_lpgbt[0]):
+                counter+=1
+            minigroups[lpgbt] = counter
+            layer = layers_connected_to_lpgbt[0]
+            #print(layers_connected_to_lpgbt[0] , counter)
```
Comment (snwebb) [inline process.py]: Remove commented code (or if necessary set it to print level 3)

File: process.py [hunk 3]
```diff
@@ -302,42 +302,71 @@ def getlpGBTHists(data, module_hists):
...
+            if(len(dict.fromkeys(layers_connected_to_lpgbt))!=1):
+                print("LpGBT connected to more than one layer. Change algorithm!!")
+
+            if (layer !=layers_connected_to_lpgbt[0]):
                 counter+=1
+            if(layer ==50 and layers_connected_to_lpgbt[0]==37):
+                counter=22
+            minigroups[lpgbt] = counter
+            layer = layers_connected_to_lpgbt[0]
+            #print(layers_connected_to_lpgbt[0] , counter)
```
Comment (snwebb) [inline process.py]: same as above comment here

**Review by snwebb — COMMENTED** [review]
The changes look good, I just had a few small comments, then we can merge.

**Comment by mhassans** [comment]
Hi Samuel,

Yes thanks please perform the changes and merge it.

**Review by snwebb — APPROVED** [review]
Request approved

**Pull request merged**